db/crud_test.go

```diff
@@ -956,3 +956,211 @@ func BenchmarkHandleRevDelta(b *testing.B) {
+	bodyBytes, removed, err = db.get1xRevFromDoc(doc1, rev1, true)
+	assert.False(t, removed)
+	assert.NoError(t, err, "It should not throw any error")
+	assert.NotNil(t, bodyBytes, "Document body bytes should be received")
+	assert.NoError(t, response.Unmarshal(bodyBytes))
+
+	// Delete the revision rev2 and get the revision from doc2
+	rev3, err := db.DeleteDoc("camera", rev2)
+	bodyBytes, removed, err = db.get1xRevFromDoc(doc2, rev3, true)
+	assert.False(t, removed)
+	assert.Error(t, err, "It should throw 404 missing error")
+	assert.Nil(t, bodyBytes, "Document body bytes should be empty")
+
+	// Get the revision details from doc2 and rev2 after rev2 is deleted
+	bodyBytes, removed, err = db.get1xRevFromDoc(doc2, "", true)
+	assert.False(t, removed)
+	assert.NoError(t, err, "It should not throw any error")
+	assert.NotNil(t, bodyBytes, "Document body bytes should be received")
+
+	assert.NoError(t, response.Unmarshal(bodyBytes))
+	log.Printf("body:%v", response)
+}
+
+func TestGet1xRevFromDoc(t *testing.T) {
+	testBucket := base.GetTestBucket(t)
+	defer testBucket.Close()
+	bucket := testBucket.Bucket
+
+	context, err := NewDatabaseContext("db", bucket, false, DatabaseContextOptions{})
+	assert.NoError(t, err, "Couldn't create context for database 'db'")
+	defer context.Close()
+	db, err := CreateDatabase(context)
+	assert.NoError(t, err, "Couldn't create database 'db'")
+
+	// Create the first revision of the document
+	payload := `{"model":"1483C002SKU","sku":"5578523","name":"Canon EOS 5D Mark IV","price":"$2,499.99","version":"1a"}`
+	doc1, rev1, err := db.PutExistingRevWithBody("camera", unjson(payload), []string{"1-a"}, false)
+	assert.NoError(t, err, "Couldn't create document")
+	assert.NotEmpty(t, doc1, "Document shouldn't be empty")
+	assert.Equal(t, "1-a", rev1, "Provided input revision ID should be returned")
+
+	// Create the second revision of the document
+	payload = `{"model":"1483C002SKU","sku":"5578523","name":"Canon EOS 5D Mark IV","price":"$2,499.99","version":"2a"}`
+	doc2, rev2, err := db.PutExistingRevWithBody("camera", unjson(payload), []string{"2-a", "1-a"}, false)
+	assert.NoError(t, err, "Couldn't create document")
+	assert.NotEmpty(t, doc2, "Document shouldn't be empty")
+	assert.Equal(t, "2-a", rev2, "Provided input revision ID should be returned")
+
+	// Get the 1x revision from second doc.
+	bodyBytes, removed, err := db.get1xRevFromDoc(doc2, rev2, true)
+	assert.NotEmpty(t, bodyBytes, "Document body bytes should be returned")
+	assert.False(t, removed, "This shouldn't be a removed document")
+	var response = Body{}
+	assert.NoError(t, response.Unmarshal(bodyBytes))
+	assert.Equal(t, "camera", response[BodyId])
+	assert.Equal(t, "2-a", response[BodyRev])
+	assert.Equal(t, "1483C002SKU", response["model"])
+	assert.Equal(t, "Canon EOS 5D Mark IV", response["name"])
+	assert.Equal(t, "$2,499.99", response["price"])
+	assert.Equal(t, "5578523", response["sku"])
```

Comment:
Can reduce the number of properties in the doc (or at least reduce the number of properties you're asserting) - only really need one or two to verify that they are different between rev1 and rev2

Reply:
Sure Adam, I have committed required changes in the latest revision.